lib/server.js

```diff
@@ -20,6 +30,12 @@ export class Http3Server extends Http3WebTransport {
 
     /** @type {Record<string, ReadableStreamController<any>>} */
     this.sessionController = {}
+
+    this._ready = defer()
```

Comment:
This adds a `_ready` field to the server.  We may wish to store these in a `WeakMap` instead to hide them from the types or better (I think - more familiar at least) switch to an `EventEmitter` model for `open`/`connect`/`close`/`end`/`error` events for the server side like the http2 server in node core.

Reply:
Well, I would like to stick with the more modern approach. Since it is similar to the sessions etc in WebTransport. But you have a point many developers may be more familiar with the old stuff and have code compatible with this. If you like you can always make a wrapper package that maps the server, client, and sessions for this package and also the browser to an old-fashioned interface. There are probably some people who like it. But I actually like the promise-based steam-style interface of webtransport, after first hating it.

Reply:
Sure I have looked at it, but it progresses slowly and it is hard to guess, what and when it comes.
But I also do not know, if the initial http/3 support has webtransport support.
So that is nothing I worry about currently.

Reply:
Also, I assume that the http/3 interface would be different from http/2 (as HTTP/2 differs from HTTP), just because the protocol itself is different.
Just out of curiosity, do you want only to use WebTransport or also http/3 (not implemented), I assume only webtransport?

Anyway, I believe, that can one make a wrapper that unpromisifies the promises and streams into callbacks.
Btw. here is an article describing the advantages of promises:
https://dev.to/somedood/promises-and-events-some-pitfalls-and-workarounds-elp
probably these things are the reason the whole webtransport API is designed as it is.